File: AndroidAvatar/src/dev/game/legend/avatar/Avatar.java
package dev.game.legend.avatar;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import dev.game.legend.Player;
import dev.game.legend.R;
import dev.game.legend.Sprite;
import dev.game.legend.data.CharInfo;

public class Avatar extends Player{

	public Avatar(Context c, CharInfo info) {
		super(c, info);
	}

	public Avatar(Context c) {
		super(c, new CharInfo(0));
	}
	@Override
	public void init(CharInfo c) {
		info=c;
		Resources res = context.getResources();
		Bitmap[] bitmaps = new Bitmap[12];
		bitmaps[0] = BitmapFactory.decodeResource(res, R.drawable.basic_head);
		bitmaps[1] = BitmapFactory.decodeResource(res, R.drawable.bat_head);
		bitmaps[2] = BitmapFactory.decodeResource(res, R.drawable.bird_head);
		bitmaps[3] = BitmapFactory.decodeResource(res, R.drawable.fish_head);
		bitmaps[4] = BitmapFactory.decodeResource(res, R.drawable.orc_head);
		bitmaps[5] = BitmapFactory.decodeResource(res, R.drawable.elf_head);
		bitmaps[6] = BitmapFactory.decodeResource(res, R.drawable.horse_head);
		bitmaps[7] = BitmapFactory.decodeResource(res, R.drawable.human_head);
		bitmaps[8] = BitmapFactory.decodeResource(res, R.drawable.lizard_head);
		bitmaps[9] = BitmapFactory.decodeResource(res, R.drawable.yinoch_head);
		bitmaps[10] = BitmapFactory.decodeResource(res, R.drawable.eden_head);
		bitmaps[11] = BitmapFactory.decodeResource(res, R.drawable.iszu_head);
		head = new Sprite(bitmaps, p2);
		
	
		bitmaps = new Bitmap[3];
		bitmaps[0] = BitmapFactory.decodeResource(res,  R.drawable.eden_face2);
		bitmaps[1] = BitmapFactory.decodeResource(res, R.drawable.face2);
		bitmaps[2] = BitmapFactory.decodeResource(res, R.drawable.iszu_face2);
		face = new Sprite(bitmaps, p2);
		
		bitmaps = new Bitmap[9];
		bitmaps[0] = BitmapFactory.decodeResource(res, R.drawable.body);
		bitmaps[1] = BitmapFactory.decodeResource(res, R.drawable.job_archer);
		bitmaps[2] = BitmapFactory.decodeResource(res, R.drawable.job_mage);
		bitmaps[3] = BitmapFactory.decodeResource(res, R.drawable.job_monk);
		bitmaps[4] = BitmapFactory.decodeResource(res, R.drawable.job_ninja);
		bitmaps[5] = BitmapFactory.decodeResource(res, R.drawable.job_wizard);
		bitmaps[6] = BitmapFactory.decodeResource(res, R.drawable.job_swordman);
		bitmaps[7] = BitmapFactory.decodeResource(res, R.drawable.job_soldier);
		bitmaps[8] = BitmapFactory.decodeResource(res, R.drawable.iszu_body);
		body = new Sprite(bitmaps, p2);

		bitmaps = new Bitmap[8];
		bitmaps[0] = BitmapFactory.decodeResource(res, R.drawable.element_air);
		bitmaps[1] = BitmapFactory.decodeResource(res, R.drawable.element_tree);
		bitmaps[2] = BitmapFactory.decodeResource(res, R.drawable.element_fire);
		bitmaps[3] = BitmapFactory.decodeResource(res, R.drawable.element_rock);
		bitmaps[4] = BitmapFactory.decodeResource(res, R.drawable.element_earth);
		bitmaps[5] = BitmapFactory.decodeResource(res, R.drawable.element_ice);
		bitmaps[6] = BitmapFactory.decodeResource(res, R.drawable.element_water);
		bitmaps[7] = BitmapFactory.decodeResource(res, R.drawable.element_thunder);
		element = new Sprite(bitmaps, p2);

		bitmaps = new Bitmap[8];
		bitmaps[0] = BitmapFactory.decodeResource(res, R.drawable.wapon_bow);
		bitmaps[1] = BitmapFactory.decodeResource(res, R.drawable.wapon_staff);
		bitmaps[2] = BitmapFactory.decodeResource(res, R.drawable.wapon_sword);
		bitmaps[3] = BitmapFactory.decodeResource(res, R.drawable.wapon_spear);
		bitmaps[4] = BitmapFactory.decodeResource(res, R.drawable.wapon_schimitar);
		bitmaps[5] = BitmapFactory.decodeResource(res, R.drawable.wapon_kane);
		bitmaps[6] = BitmapFactory.decodeResource(res, R.drawable.wapon_chain);
		bitmaps[7] = BitmapFactory.decodeResource(res, R.drawable.wapon_shield);
		wapon = new Sprite(bitmaps, p2);

		// cache handles to our key sprites & other drawables
		bitmaps = new Bitmap[1];
		bitmaps[0] = BitmapFactory.decodeResource(res,  R.drawable.basic_hand1);
		lFist = new Sprite(bitmaps, p2);
		
		bitmaps = new Bitmap[1];
		bitmaps[0] = BitmapFactory.decodeResource(res, R.drawable.basic_hand6);
		rFist = new Sprite(bitmaps, p2);
		
		bitmaps= new Bitmap[1];
		bitmaps[0] = BitmapFactory.decodeResource(res,  R.drawable.basic_arm1);
		lArm = new Sprite(bitmaps, p2);
		
		bitmaps = new Bitmap[1];
		bitmaps[0] = BitmapFactory.decodeResource(res,  R.drawable.basic_arm5);
		rArm = new Sprite(bitmaps, p2);
		
		legs = new Sprite(BitmapFactory.decodeResource(res,	  R.drawable.basic_legs) , p2);
		feets = new Sprite(BitmapFactory.decodeResource(res,  R.drawable.basic_feets), p2);
		icon = new Sprite(BitmapFactory.decodeResource(res,	  R.drawable.basic_icon) , p2);
		skill = new Sprite(BitmapFactory.decodeResource(res,  R.drawable.job)  , p2);
	}
	
	@Override
	public void initPosition(){
		super.initPosition();
		rFist.setPosition(80, 240);
		rArm.setFrame(1);
		lArm.setFrame(1);
		lFist.setFrame(0);
		rFist.setFrame(2);
	}
	
	@Override
	public 	void drawImage(Canvas c){
		super.drawImage(c);
		//lFist.draw(c);
		rFist.draw(c);
		element.draw(c);
		wapon.draw(c);
		
	}
	@Override
	public void move(int x,int y){
		super.move(x, y);
		element.move(x, y);
		wapon.move(x,y);
		rFist.move(x,y);
		
	}
}

File: iPhoneAvatar/Classes/Avatar.m
//
//  Avatar m
//  Avatar
//
//  Created by Jin Hu on 2010/08/04 
//  Copyright 2010 __MyCompanyName__  All rights reserved 
//

#import "Avatar.h"
#import "CharInfo.h"
#import "Sprite.h"
@implementation Avatar
//@synthesize  head, face,body,legs,feets,element,wapon;


-(Sprite*)initPart:(NSString*)part {
	NSMutableArray* bitmaps = [NSMutableArray arrayWithCapacity:12];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"bird_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"lizard_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"orc_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"bat_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"basic_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"elf_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"fish_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"horse_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"human_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"iszu_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"yinoch_%@.png",part]]];
	[bitmaps addObject:[UIImage imageNamed:[NSString stringWithFormat: @"eden_%@.png",part]]];
	Sprite* s = [[Sprite alloc] initWithArray:bitmaps mirror:p2]; 
	return s;
}

-(id) init:(CharInfo *)c{
	NSMutableArray* bitmaps ;
	head = [self initPart:@"head"];
	face = [self initPart:@"face_normal"];
	body = [self initPart:@"body"];
	lFist = [self initPart:@"hand_l_rock"];
	rFist = [self initPart:@"hand_r_paper"];
	lArm = [self initPart:@"arm_l1"];
	rArm = [self initPart:@"arm_r2"];
	body = [self initPart:@"body"];
	legs = [self initPart:@"legs"];
	feets = [self initPart:@"feets"];
	
	bitmaps = [NSMutableArray arrayWithCapacity:12];
	[bitmaps addObject:[UIImage imageNamed:@"job.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_archer.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_mage.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_monk.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_ninja.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_wizard.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_swordman.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"job_soldier.png"]];
	skill = [[Sprite alloc] initWithArray:bitmaps mirror:p2]; 

	bitmaps = [NSMutableArray arrayWithCapacity:12];
	[bitmaps addObject:[UIImage imageNamed:@"element.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_air.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_tree.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_fire.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_rock.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_earth.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_ice.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_water.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"element_thunder.png"]];
	element = [[Sprite alloc] initWithArray:bitmaps mirror:p2]; 
	
	bitmaps =[NSMutableArray arrayWithCapacity:12];
	[bitmaps addObject:[UIImage imageNamed:@"wapon.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_bow.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_staff.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_sword.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_spear.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_schimitar.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_kane.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_chain.png"]];
	[bitmaps addObject:[UIImage imageNamed:@"wapon_shield.png"]];
	wapon = [[Sprite alloc] initWithArray:bitmaps mirror:p2]; 
	
	
	icon = [[Sprite alloc] initWithImage:[UIImage imageNamed:@"basic_icon.png" ] mirror:p2]; 

	[self initPosition];
	return self;
}


- (void)dealloc {
    [super dealloc];
}

-(void) initPosition{
	[super initPosition];
	[rFist setPosition:80 y:240];
	[rArm setCurrentFrame:1];
	[lArm setCurrentFrame:1];
	[lFist setCurrentFrame:0];
	[rFist setCurrentFrame:2];
}

-(void) drawImage{
	[super drawImage];
	[rFist paint];
	[element paint];
	[wapon paint ];
	[lFist paint];
	
}
-(void) move:(int) x2 y:(int) y2{
	[super move:x2 y:y2];
	[element move:x2 y:y2];
	[wapon move:x2 y:y2];
	[rFist move:x2 y:y2];
	[lFist move:-55 y:30];
}
-(void)setAllFrame:(int)frame{
	frame = frame%12;
	[head setCurrentFrame:frame];
	[face setCurrentFrame:frame];
	[body setCurrentFrame:frame];
	[lArm setCurrentFrame:frame];
	[rArm setCurrentFrame:frame];
	[lFist setCurrentFrame:frame];
	[rFist setCurrentFrame:frame];
	[legs setCurrentFrame:frame];
	[feets setCurrentFrame:frame];
//	[rArm setFrame:frame];
//	[rArm setFrame:frame];
//	[rArm setFrame:frame];
//	[rArm setFrame:frame];
}
@end

